handlers/people_test.go

```diff
@@ -258,3 +258,191 @@ func TestDeletePerson(t *testing.T) {
 		mockDb.AssertExpectations(t)
 	})
 }
+
+func TestGetPeopleBySearch(t *testing.T) {
+	mockDb := mocks.NewDatabase(t)
+	pHandler := NewPeopleHandler(mockDb)
+
+	t.Run("should return users that match the search text", func(t *testing.T) {
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(pHandler.GetPeopleBySearch)
+		expectedPeople := []db.Person{
+			{ID: 1, Uuid: "uuid1", OwnerPubKey: "pubkey1", OwnerAlias: "John Doe"},
+			{ID: 2, Uuid: "uuid2", OwnerPubKey: "pubkey2", OwnerAlias: "John Smith"},
+		}
+
+		rctx := chi.NewRouteContext()
+		rctx.URLParams.Add("search", "John")
+		req, err := http.NewRequestWithContext(context.WithValue(context.Background(), chi.RouteCtxKey, rctx), http.MethodGet, "/search", nil)
+		assert.NoError(t, err)
+
+		mockDb.On("GetPeopleBySearch", mock.Anything).Return(expectedPeople)
+		handler.ServeHTTP(rr, req)
+
+		var returnedPeople []db.Person
+		err = json.Unmarshal(rr.Body.Bytes(), &returnedPeople)
+		assert.NoError(t, err)
+		assert.Equal(t, http.StatusOK, rr.Code)
+		assert.EqualValues(t, expectedPeople, returnedPeople)
+		mockDb.AssertExpectations(t)
+	})
+
+	t.Run("should return an empty search result when no user matches the search text", func(t *testing.T) {
+		mockDb.ExpectedCalls = nil
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(pHandler.GetPeopleBySearch)
+		expectedPeople := []db.Person{}
+
+		rctx := chi.NewRouteContext()
+		rctx.URLParams.Add("search", "user not matched")
```

Comment:
@AbdulWahab3181 the search should be in the URL query, not param.

Reply:
Addressed